tests/unit/s2n_ktls_test.c

```diff
@@ -13,8 +13,45 @@
  * permissions and limitations under the License.
  */
 
-#include "crypto/s2n_cipher.h"
+#include "tls/s2n_ktls.h"
+
 #include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+S2N_RESULT s2n_ktls_retrieve_file_descriptor(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, int *fd);
+
+/* set kTLS supported cipher */
+struct s2n_cipher ktls_temp_supported_cipher = {
+    .ktls_supported = true,
+};
+struct s2n_record_algorithm ktls_temp_supported_record_alg = {
+    .cipher = &ktls_temp_supported_cipher,
+};
+struct s2n_cipher_suite ktls_temp_supported_cipher_suite = {
+    .record_alg = &ktls_temp_supported_record_alg,
+};
+
+S2N_RESULT s2n_test_configure_mock_ktls_connection(struct s2n_connection *conn, int fd, bool complete_handshake)
```

Comment:
This threw me a bit: If you're trying to test configuring a connection for ktls, why are you mocking the configuration? A better name might be "s2n_setup_valid_connection_for_ktls". I would also suggest removing the "fd" and "complete_handshake" flags-- you never actually use "fd" for anything, and "complete_handshake" is the only option to produce a connection that's not valid for ktls. Instead, you can just set conn->handshake.message_number = 0 to reset the handshake.